src/main/java/org/frankframework/frankdoc/AttributeReuseManager.java

```diff
@@ -0,0 +1,197 @@
+				callback.addReusableAttribute(attributeReference.frankAttribute);
+				definedReusableAttributes.add(attributeReference.frankAttribute);
+			}
+			callback.addReusedAttributeReference(attributeReference.frankAttribute, attributeReference.xsdGroup, attributeReference.xsdGroupName);
+		} else {
+			callback.addAttributeInline(attributeReference.frankAttribute, attributeReference.xsdGroup, attributeReference.xsdGroupName);
+		}		
+	}
+
+	private static class ReferencedFrankAttribute {
+		private List<AttributeReference> itemsThatShareFrankAttribute = new ArrayList<>();
+
+		ReferencedFrankAttribute(AttributeReference first) {
+			itemsThatShareFrankAttribute.add(first);
+		}
+
+		void add(AttributeReference item) {
+			if(item.frankAttribute != itemsThatShareFrankAttribute.get(0).frankAttribute) {
+				throw new IllegalArgumentException("Referenced FrankAttribute does not match");
+			}
+			itemsThatShareFrankAttribute.add(item);
+		}
+
+		boolean isFrankAttributeReferencedOnce() {
+			return itemsThatShareFrankAttribute.size() == 1;
+		}
+
+		String getAttributeName() {
+			return itemsThatShareFrankAttribute.get(0).frankAttribute.getName();
+		}
+
+		FrankAttribute getFrankAttribute() {
+			return itemsThatShareFrankAttribute.get(0).frankAttribute;
+		}
+
+		void setNoAttributeReferenceReuses() {
+			itemsThatShareFrankAttribute.stream().forEach(item -> item.reused = false);
+		}
+	}
+
+	private static class ReferencedFrankAttributeNameGroup {
+		private Map<FrankAttribute, ReferencedFrankAttribute> referencedFrankAttributes = new HashMap<>();
+
+		ReferencedFrankAttributeNameGroup(AttributeReference first) {
+			ReferencedFrankAttribute referencedFrankAttribute = new ReferencedFrankAttribute(first);
+			referencedFrankAttributes.put(first.frankAttribute, referencedFrankAttribute);
+		}
+
+		private String getName() {
+			return referencedFrankAttributes.values().iterator().next().getAttributeName();
+		}
+
+		void add(AttributeReference item) {
+			if(! item.frankAttribute.getName().equals(getName())) {
+				throw new IllegalArgumentException("FrankAttribute name mismatch");
+			}
+			if(referencedFrankAttributes.containsKey(item.frankAttribute)) {
+				referencedFrankAttributes.get(item.frankAttribute).add(item);
+			} else {
+				ReferencedFrankAttribute referencedFrankAttribute = new ReferencedFrankAttribute(item);
+				referencedFrankAttributes.put(item.frankAttribute, referencedFrankAttribute);
+			}
+		}
+
+		List<ReferencedFrankAttribute> getAttributesReferencedOnce() {
+			return referencedFrankAttributes.values().stream()
+					.filter(ReferencedFrankAttribute::isFrankAttributeReferencedOnce)
+					.collect(Collectors.toList());
+		}
+
+		boolean hasMultipleReusedAttributes() {
+			return referencedFrankAttributes.values().stream()
+					.filter(g -> ! g.isFrankAttributeReferencedOnce())
+					.map(ReferencedFrankAttribute::getFrankAttribute)
+					.distinct()
+					.collect(Collectors.counting()) != 1L;
```

Comment:
Hier zou je toch willen dat als de count boven de 1 komt de loop wordt afgebroken.

Reply:
Maar ik bedoel dat je niet precies hoeft te weten wat de count is. Je hebt aan '0', '1' of 'meer dan 1' genoeg. Als je teller twee is geworden, dan hoef je niet verder te tellen.
Ik denk dat je iets als `allMatch(count<2)` zou moeten kunnen gebruiken

Reply:
Leuk! Ik heb een manier gevonden die aan je verzoek beantwoordt.